server/ports.go

```diff
@@ -0,0 +1,81 @@
+package server
+
+import (
+	"fmt"
+	"sync"
+)
+
+// portManager is responsible for maintaining segregated access to the port field
+// via the use of rwLock sync.RWMutex and mustRead sync.Mutex
+// rwLock establishes a standard read write mutex that allows consecutive reads and exclusive writes
+// mustRead forces MustGetPort to wait until port has a none 0 value
+type portManger struct {
```

Comment:
Yes, it ideally needs to be an internal struct and functions. It could be exported but I can't think of any other part of the application that would use this specific set of functionality.

Also I wanted the `portManager` to exist to be the only place that port handling occurred, so it was clear that only `portManager` should do port related activity.

But I 100% do not want any dependent code trying to interact with the internal workings of `portManager` because this will definitely lead to deadlocks and infinite hangs.